dkg/dkg_test.go

```diff
@@ -405,3 +408,134 @@ func startNewDKG(t *testing.T, parentCtx context.Context, config dkg.Config, dkg
 
 	return cancel
 }
+
+// TestKeymanager tests the keymanager functionality where the dkg command pushes validator keyshares
+// to a keymanager API url.
+func TestKeymanager(t *testing.T) {
+	numNodes := 4
+	lock, p2pKeys, _ := cluster.NewForT(t, 1, 3, numNodes, 0)
+	def := lock.Definition
+	require.NoError(t, def.VerifySignatures())
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Setup
+	dir, err := os.MkdirTemp("", "")
+	require.NoError(t, err)
+
+	allReceived := make(chan struct{}, 1) // Receives struct{} if all `numNodes` keystores are intercepted by the keymanager server
```

Comment:
why buffer of 1? unbuffered is default

Reply:
i initially intended to make it unblocking, but i have updated the tests and made this unbuffered